File: src/nkc/Bot.java
package nkc;

import java.awt.geom.Point2D;

import robocode.AdvancedRobot;
import robocode.ScannedRobotEvent;

public class Bot {
	public double bearingRadians;
	public double headingRadians;
	public double absoluteBearing;
	public double distance;
	public String name;
	public double energy;
	public double heading;
	public double velocity;
	public double x;
	public double y;
	
	
	static Bot fromScannedRobotEvent(ScannedRobotEvent e, AdvancedRobot scanner) {
		Bot bot = new Bot();
		bot.energy = e.getEnergy();
		bot.velocity = e.getVelocity();
		bot.heading = e.getHeading();
		bot.bearingRadians = e.getBearingRadians();
		bot.distance = e.getDistance();
		bot.name = e.getName();
		bot.headingRadians = e.getHeadingRadians();
		bot.absoluteBearing = scanner.getHeadingRadians() + e.getBearingRadians();
		
		bot.x = scanner.getX() + e.getDistance() * Math.sin(bot.absoluteBearing);
		bot.y = scanner.getY() + e.getDistance() * Math.cos(bot.absoluteBearing);
		
		return bot;
	}
	
	Point2D getLocation() {
		return new Point2D.Double(x, y);
	}
}

File: src/NKC.RobotsOfDeath/NKC.RobotsOfDeath/Bot.cs
using NKC.RobotsOfDeath.Maths;
using Robocode;
using System;
using System.Collections.Generic;
using System.Text;

namespace NKC.RobotsOfDeath
{
    public class ScannedBot
    {
        public double bearingRadians;
        public double headingRadians;
        public double absoluteBearing;
        public double distance;
        public String name;
        public double energy;
        public double heading;
        public double velocity;
        public double x;
        public double y;
        public double bearing;

        AdvancedRobot scanner;


        public ScannedBot(ScannedRobotEvent e, AdvancedRobot scanner)
        {
            this.scanner = scanner;
            Update(e);
        }

        public void Update(ScannedRobotEvent e)
        {
            bearing = e.Bearing;
            energy = e.Energy;
            velocity = e.Velocity;
            heading = e.Heading;
            bearingRadians = e.BearingRadians;
            distance = e.Distance;
            name = e.Name;
            headingRadians = e.HeadingRadians;
            absoluteBearing = scanner.HeadingRadians + e.BearingRadians;

            x = scanner.X + e.Distance * Math.Sin(absoluteBearing);
            y = scanner.Y + e.Distance * Math.Cos(absoluteBearing);
        }

        Point getLocation()
        {
            return new Point(x, y);
        }
    }
}
